src/components/MediaCreationModal/MediaCreationModal.tsx

```diff
@@ -0,0 +1,350 @@
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+  Text,
+  Icon,
+  useModalContext,
+  Progress,
+  ListItem,
+  UnorderedList,
+  VStack,
+  Flex,
+} from "@chakra-ui/react"
+import { Button, Infobox } from "@opengovsg/design-system-react"
+import { useEffect, useState } from "react"
+import { FileRejection } from "react-dropzone"
+import { BiCheckCircle, BiSolidErrorCircle } from "react-icons/bi"
+import { useParams } from "react-router-dom"
+
+import { Attachment } from "components/Attachment"
+
+import { useCreateMultipleMedia } from "hooks/mediaHooks/useCreateMultipleMedia"
+
+import { MediaDirectoryParams } from "types/folders"
+
+import { Dropzone } from "./components/Dropzone"
+
+type MediaSteps = "upload" | "progressing" | "success" | "failed"
+type UploadVariant = "files" | "images"
+
+const IMAGE_UPLOAD_ACCEPTED_MIME_TYPES = {
+  "image/jpeg": [".jpg", ".jpeg"],
+  "image/png": [".png"],
+  "image/gif": [".gif"],
+  "image/svg+xml": [".svg"],
+  "image/tiff": [".tiff", ".tif"],
+  "image/bmp": [".bmp"],
+}
+
+const FILE_UPLOAD_ACCEPTED_MIME_TYPES = {
+  "application/pdf": [".pdf"],
+}
+
+interface MediaDropzoneProps {
+  fileRejections: FileRejection[]
+  uploadedFiles: File[]
+  setUploadedFiles: (files: File[]) => void
+  setFileRejections: (rejections: FileRejection[]) => void
+  isDisabled?: boolean
+  onUpload: (files: File[]) => Promise<void>
+  variant: UploadVariant
+}
+
+const MediaDropzone = ({
+  fileRejections,
+  uploadedFiles,
+  setUploadedFiles,
+  setFileRejections,
+  isDisabled,
+  onUpload,
+  variant,
+}: MediaDropzoneProps) => {
+  const { onClose } = useModalContext()
+
+  return (
+    <>
+      <ModalHeader>Upload files</ModalHeader>
```

Comment:
suggestion(ux, non-blocking): should use `getMediaLabels()` to get the appropriate labels here, since this is shared between images and files.

Reply:
will update

Reply:
actually i asked our designer for a [copy](https://opengovproducts.slack.com/archives/CK68JNFHR/p1699349295935939?thread_ts=1699331583.799259&cid=CK68JNFHR) that worked across files/images so not inclined to do this tbh @__@